dali/sasha/saarch.cpp

```diff
@@ -85,213 +85,343 @@ static void mkDateCompare(bool dfu,const char *dt,StringBuffer &out,char fill)
-                                        overflowed = true;
-                                        break;
-                                    }
-                                }
-                                else {
-                                    cmd->addId(name.str());
-                                    if (hasdtoutput) {
-                                        CDateTime dt;
-                                        di2->getModifiedTime(dt);
-                                        cmd->addDT(dt);
-                                    }
-                                }
-                                if (cmd->getAction()==SCA_GET) {
-                                    StringBuffer xml;
-                                    toXML(t,xml);
-                                    if (!cmd->addResult(xml.str()))
-                                        break;
-                                }
-                            }
                         }
                     }
-                    if (index>start+num)
+
+                    if (sortInc)
+                    {//The output should contain WUs from new to old. So, we need to reverse the order before output.
+                        Owned<CWUData> wu = new CWUData();
+                        if (outputXML || !isEmptyString(outputFormat))
+                            wu->wuTree.setown(wuTree.getClear());
+                        if (outputModifiedTime)
+                            fileIterator->getModifiedTime(wu->modifiedTime);
+                        wus.append(*wu.getClear());
+                    }
+                    else
+                    {
+                        CDateTime dt;
+                        if (outputModifiedTime)
+                            fileIterator->getModifiedTime(dt);
+
+                        if (!setOutput(cmd, outputFields, wuTree, dt))
+                        {
+                            overflowed = true;
+                            break;
+                        }
+                    }
+                    wuCount++;
+                    if (wuCount == numWUs)
                         break;
                 }
+                if (overflowed || (wuCount == numWUs))
+                    break;
             }
+            if (sortInc)
+            {
+                ForEachItemInRev(i, wus)
+                {
+                    CWUData &wuData = wus.item(i);
+                    if (!setOutput(cmd, outputFields, wuData.wuTree, wuData.modifiedTime))  //Log an error?
+                        break;
+                }
+            }
+
+            if (hasWUSOutput)
+                cmd->setWUSresult(WUSbuf);
         }
-    }
-    if (cmd->getOnline()) {
-        if (haswusoutput)
+    };
+
+    Owned<CWUReader> reader = new CWUReader(cmd, mask);
+    if (!cmd->getOnline())
+        reader->getWUs(cmd, cmd->queryBeforeWU(), cmd->queryAfterWU(), cmd->getLimit());
+    else
```

Comment:
old code specifically check if (cmd->getOnline()) here. (and not else)
There was the implication in the implementation, that both at some point could be supported.
Unless there's a good reason not to allow a single search (at some point) with both online and archived, I think should leave implementation that could handle both as was.